terraform/modules/forest_node/user-data.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+# Create a login user
+adduser --home "/home/${NEW_USER}" --disabled-password --gecos "" "${NEW_USER}"
+
+# Create .ssh directory for created user
+mkdir --parents -- "/home/${NEW_USER}/.ssh"
+chown "${NEW_USER}:${NEW_USER}" "/home/${NEW_USER}/.ssh"
+chmod 0700 "/home/${NEW_USER}/.ssh"
+
+# Check if authorized_keys file exists and copy it from root to new user
+if [ -f "/root/.ssh/authorized_keys" ]; then
+  cp /root/.ssh/authorized_keys "/home/${NEW_USER}/.ssh/authorized_keys"
+  chown "${NEW_USER}:${NEW_USER}" "/home/${NEW_USER}/.ssh/authorized_keys"
+  chmod 0600 "/home/${NEW_USER}/.ssh/authorized_keys"
+fi
+
+# Allow created user to log in
+echo "AllowUsers ${NEW_USER}" >> /etc/ssh/sshd_config
+
+# Restart SSHD
+systemctl restart sshd
+
```

Comment:
```suggestion
: Allow user to log in via ssh
echo "AllowUsers ${NEW_USER}" >> /etc/ssh/sshd_config
systemctl restart sshd

```

A lot of your comments so far are of the form
```bash
# do the thing
do-the-thing
```

Which isn't very helpful - comments are to help the reader:
- understand the layout of the script (large sets of related actions which)
- answer reader questions ahead of time (less obvious knowledge)

the line `systemctl restart sshd` _already tells me_ that you're restarting sshd, you don't need to do that twice